### DISCUSSION TITLE
put react-query's result to store like zustand

### DISCUSSION THREAD

**Discussion by wyyufeng** [TanStack/query#7490] [Q&A]
I've read some articles like [https://tkdodo.eu/](https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose#ondatachanged) and I know that storing the results of react-query in a store (like zustand) is not agood practice。
but in some scenarios, you need to do some state updates based on the server state, and then submit them to the server, in this case, is it appropriate to store the server state in the store, or is it appropriate to use an API like setQueryData to update the server state
 here's an example [https://codesandbox.io/p/sandbox/loving-shadow-yqsv9s]( https://codesandbox.io/p/sandbox/loving-shadow-yqsv9s)

**Comment by TkDodo** [comment]
I'm covering this here: https://tkdodo.eu/blog/react-query-and-forms

**Reply by wyyufeng** [reply]
Thanks for the reply, I read the article carefully and I think there can be a better solution in terms of using react-hook-form to achieve integration with forms, but the one in my scenario may not be so straightforward (it's a more complex workflow scenario).
My point is that as long as one understands the pros and cons of each scenario, synchronizing the results to the store is not entirely undesirable!
